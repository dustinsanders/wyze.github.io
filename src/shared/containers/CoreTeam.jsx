// @flow

import type { Style, Team, Teams, VNode } from '../../types'
import { css, media, select } from 'glamor'
import { small } from '../styles/breakpoints'
import Box from '../components/Box'
import Link from '../components/Link'
import Section from '../components/Section'
import preact from 'preact'

type Props = {
  teams: Teams,
}

const core: Style = css(
  {
    paddingTop: '.4em',
    width: '100%',
  },
  select('& + &', { paddingTop: '1em' }),
  media(small, css(
    {
      width: '50%',
    },
    select('& + &', { paddingTop: 0 }),
  )),
)

const link: Style = css({
  display: 'inline-block',
  height: '5.7em',
  minWidth: '5.7em',
})

const CoreTeam = ({ teams }: Props): VNode => (
  <Box wrap title="Core Team Member">
    {teams.map(({ component: Component, to }: Team): VNode => (
      <Section key={to} styles={core}>
        <Link styles={link} to={to}>
          <Component />
        </Link>
      </Section>
    ))}
  </Box>
)

export default CoreTeam
