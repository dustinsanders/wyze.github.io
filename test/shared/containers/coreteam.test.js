// @flow

import CoreTeam from '../../../src/shared/containers/CoreTeam'
import preact from 'preact'
import snapshot from '../../helpers/snapshot'

const teams = [
  // eslint-disable-next-line react/jsx-filename-extension
  { component: <svg />, name: 'My Team' },
]

describe('<CoreTeam />', () => {
  it('renders', snapshot.bind(null, CoreTeam, { teams }))
})
