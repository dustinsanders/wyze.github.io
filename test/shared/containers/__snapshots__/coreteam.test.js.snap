// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoreTeam /> renders 1`] = `
"<Box wrap title=\\"Core Team Member\\">
  <Section styles=\\"css-1y4bstf\\">
    <Link styles=\\"css-5j31gr\\">
      <[object Object]></[object Object]>
    </Link>
  </Section>
</Box>"
`;
